### PR TITLE
Enable ToolTaskThatTimeoutAndRetry test after change powershell.exe to cmd.exe

### PR DESCRIPTION
**Pull request by JaynieBai**
Fixes [#8750](https://github.com/dotnet/msbuild/issues/8750)

### Context
[PowerShell](https://github.com/dotnet/msbuild/blob/main/src/Utilities.UnitTests/ToolTask_Tests.cs#L902) sleep is used for windows - that however prooves to be a bit unstable in CI

### Changes Made
Use cmd.exe 

### Testing
/azp run for several times. It passed.

### CODE REVIEW

**Review by JanKrivanek — APPROVED** [review]
Thanks!

File: src/Utilities.UnitTests/ToolTask_Tests.cs [hunk 1]
```diff
@@ -886,8 +886,8 @@ public void ToolTaskThatTimeoutAndRetry(int repeats, int initialDelay, int follo
         /// </remarks>
         private sealed class ToolTaskThatSleeps : ToolTask
         {
-            // PowerShell command to sleep:
-            private readonly string _powerShellSleep = "-NoProfile -ExecutionPolicy RemoteSigned -Command \"Start-Sleep -Milliseconds {0}\"";
+            // Windows command to sleep:
```
Comment (Forgind) [inline src/Utilities.UnitTests/ToolTask_Tests.cs:889]:
```suggestion
            // Windows command prompt command to sleep:
```

**Review by Forgind — APPROVED** [review]
Nice!

File: src/Utilities.UnitTests/ToolTask_Tests.cs (hunk 1 above)
Comment (JaynieBai) [reply src/Utilities.UnitTests/ToolTask_Tests.cs]: Done

**Pull request merged**